### PR TITLE
Bug 1910581: CSO shouldn't overwrite clustercsidriver objects

### PR DESCRIPTION
**Pull request by bertinatto**
With this PR CSO will no longer update/patch the CR.

If the CR doesn't exist, CSO will create it. However, if the already CR exists, CSO won't update it with the config that it sees fit.

This is a requirement for the CSI Config Observer Controller used in CSI driver operators. That controller watches `proxies` resources and writes the observed config in `CR.Spec.observedConfig`. Without this PR, CSO would overwrite that field.

CC @openshift/storage

### CODE REVIEW

**Comment by bertinatto** [comment]
/wip
Still checking if this patch fixes the issue.

File: pkg/operator/csidriveroperator/crcontroller.go [hunk 1]
```diff
@@ -200,25 +178,7 @@ func (c *CSIDriverOperatorCRController) applyClusterCSIDriver(requiredOriginal *
 		return nil, false, err
 	}
 
-	modified := resourcemerge.BoolPtr(false)
-	existingCopy := existing.DeepCopy()
-
-	resourcemerge.EnsureObjectMeta(modified, &existingCopy.ObjectMeta, required.ObjectMeta)
```
Comment (gnufied) [inline pkg/operator/csidriveroperator/crcontroller.go:206]: I think we might want to merge metadata of an object in-case there are some cluster-wide annotations applied to these CRs, but if we come to that we can also push and update and do metadata merging later on (so I am bit torn).

**Pull request merged**